src/view/view.cpp

```diff
@@ -793,13 +793,23 @@ void wf::view_interface_t::set_decoration(surface_interface_t *frame)
     if (!frame)
     {
         damage();
+
+        // Take wm geometry as it was with the decoration.
+        const auto wm = get_wm_geometry();
+
         if (view_impl->decoration)
         {
             this->remove_subsurface(view_impl->decoration);
         }
 
         view_impl->decoration = nullptr;
         view_impl->frame = nullptr;
+
+        if ((fullscreen || this->tiled_edges) && (wm != get_wm_geometry()))
```

Comment:
I think it is safe to only check whether geometry changes (after all, if we're fullscreen, the decoration should have size 0, in which case `wm == get_wm_geometry` anyway)

Reply:
This would expand normal floating views though. Non-tiling/fullscreen views don't get shrunken down on decorate in L842.. Or am I reading the code wrong?

Reply:
> This would expand normal floating views though. Non-tiling/fullscreen views don't get shrunken down on decorate in L842.. Or am I reading the code wrong?

Hmm, I can see where you're coming from. Well, for fullscreen views it doesn't make sense to change anything at all, so maybe `!fullscreen && tiled_edges && ...`?

Reply:
And if this is good for you, do leave a comment with the rationale for these conditions :)

Reply:
Alright so if I understand correctly, fullscreen views should not be decorated? (or at least their expanded wm geo would be the same as their view geo)

Reply:
Yes